### PR TITLE
[stable-v2.12] rimage: fix resource leak

### PR DESCRIPTION
**Pull request by abonislawski**

### CODE REVIEW

**Comment by kv2019i** [comment]
@abonislawski Can we merge to main first and then backport with a "cherry-picked xxx) entry? We still have time until January, so I'd prefer to have all changes go through main.

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
Let's go with direct PR here then. @abonislawski please submit to main still on the side.

File: tools/rimage/src/manifest.c [hunk 1]
```diff
@@ -512,6 +512,8 @@ static int man_module_create_reloc(struct image *image, struct manifest_module *
 		unsigned int j;
 
 		strncpy(name, (char *)sof_mod->module.name, SOF_MAN_MOD_NAME_LEN);
```
Comment (ujfalusi) [inline tools/rimage/src/manifest.c:515]: s/strncpy/strlcpy ?

Comment (lgirdwood) [reply tools/rimage/src/manifest.c:515]: This would be a further update if needed, I think we are good to go today with @abonislawski fix.

**Review by lgirdwood — APPROVED** [review]

**Comment by abonislawski** [comment]
> @abonislawski please submit to main still on the side.

Main branch PR: https://github.com/thesofproject/sof/pull/9747

**Comment by lgirdwood** [comment]
MTL failure unrelated. Looks like IGK DUT config.

**Pull request merged**